packages/expo/cli/start/doctor/dependencies/validateDependenciesVersions.ts

```diff
@@ -0,0 +1,136 @@
+  const incorrectDeps = findIncorrectDependencies(packageVersions, bundledNativeModules);
+
+  if (incorrectDeps.length > 0) {
+    Log.warn('Some dependencies are incompatible with the installed expo package version:');
+    incorrectDeps.forEach(({ packageName, expectedVersionOrRange, actualVersion }) => {
+      Log.warn(
+        ` - ${chalk.underline(packageName)} - expected version: ${chalk.underline(
+          expectedVersionOrRange
+        )} - actual version installed: ${chalk.underline(actualVersion)}`
+      );
+    });
+
+    Log.warn(
+      'Your project may not work correctly until you install the correct versions of the packages.\n' +
+        chalk`Install individual packages by running {inverse expo install [package-name ...]}`
+    );
+    // Log.warn(
+    //   'Your project may not work correctly until you install the correct versions of the packages.\n' +
+    //     `To install the correct versions of these packages, please run: ${chalk.inverse(
+    //       'expo doctor --fix-dependencies'
+    //     )},\n` +
+    //     `or install individual packages by running ${chalk.inverse(
+    //       'expo install [package-name ...]'
+    //     )}`
+    // );
+
+    return false;
+  }
+  return true;
+}
+
+function getPackagesToCheck(
+  dependencies: Record<string, string> | null | undefined,
+  bundledNativeModules: BundledNativeModules
+): string[] {
+  const dependencyNames = Object.keys(dependencies ?? {});
+  const result: string[] = [];
+  for (const dependencyName of dependencyNames) {
+    if (dependencyName in bundledNativeModules) {
+      result.push(dependencyName);
+    }
+  }
+  return result;
+}
+
+async function resolvePackageVersionsAsync(
+  projectRoot: string,
+  packages: string[]
+): Promise<Record<string, string>> {
+  const packageVersionsFromPackageJSON = await Promise.all(
+    packages.map((packageName) => getPackageVersionAsync(projectRoot, packageName))
+  );
+  return packages.reduce((acc, packageName, idx) => {
+    acc[packageName] = packageVersionsFromPackageJSON[idx];
+    return acc;
+  }, {} as Record<string, string>);
+}
+
+async function getPackageVersionAsync(projectRoot: string, packageName: string): Promise<string> {
+  let packageJsonPath: string | undefined;
+  try {
+    packageJsonPath = resolveFrom(projectRoot, `${packageName}/package.json`);
+  } catch (error: any) {
+    // This is a workaround for packages using `exports`. If this doesn't
+    // include `package.json`, we have to use the error message to get the location.
+    if (error.code === 'ERR_PACKAGE_PATH_NOT_EXPORTED') {
+      packageJsonPath = error.message.match(/("exports"|defined) in (.*)$/i)?.[2];
+    }
+  }
+  if (!packageJsonPath) {
+    throw new CommandError(
+      `"${packageName}" is added as a dependency in your project's package.json but it doesn't seem to be installed. Please run "yarn" or "npm install" to fix this issue.`
```

Comment:
Is it worth using `PackageManager.shouldUseYarn()` here to only show the relevant package manager instruction? We already check it in the methods that's (ultimately) calling this https://github.com/expo/expo/pull/16329/files#diff-19d84098bf8dbf5321b58dc3ba316a1fd5b16cdbc3f554d3ab8b0b9fa9599118R38

Reply:
Seems like the increased complexity doesn't add much value since we'd need to test against it.